cblox/include/cblox/mesh/submap_mesher.h

```diff
@@ -0,0 +1,125 @@
+#ifndef CBLOX_MESH_SUBMAP_MESHER_H_
+#define CBLOX_MESH_SUBMAP_MESHER_H_
+
+#include <memory>
+#include <vector>
+
+#include <voxblox/core/tsdf_map.h>
+#include <voxblox/integrator/merge_integration.h>
+#include <voxblox/integrator/tsdf_integrator.h>
+#include <voxblox/io/mesh_ply.h>
+#include <voxblox/mesh/mesh_integrator.h>
+#include <voxblox/mesh/mesh_layer.h>
+
+#include "cblox/core/common.h"
+#include "cblox/core/submap_collection.h"
+#include "cblox/core/tsdf_submap.h"
+
+namespace cblox {
+
+using voxblox::Mesh;
+using voxblox::MeshLayer;
+using voxblox::MeshIntegrator;
+using voxblox::MeshIntegratorConfig;
+using voxblox::Color;
+using voxblox::VertexIndex;
+
+class SubmapMesher {
+ public:
+  typedef std::shared_ptr<SubmapMesher> Ptr;
+  typedef std::shared_ptr<const SubmapMesher> ConstPtr;
+
+  // Constructor
+  SubmapMesher(const TsdfMap::Config &submap_config,
+               const MeshIntegratorConfig &mesh_config)
+      : tsdf_map_config_(submap_config), mesh_config_(mesh_config) {}
+
+  // Generating various meshes
+  template <typename SubmapType>
+  void generateSeparatedMesh(
+      const SubmapCollection<SubmapType> &submap_collection,
+      MeshLayer *seperated_mesh_layer_ptr);
+  template <typename SubmapType>
+  void generateCombinedMesh(
+      const SubmapCollection<SubmapType> &submap_collection,
+      MeshLayer *combined_mesh_layer_ptr);
+  void generatePatchMeshes(
+      const SubmapCollection<TsdfSubmap> &tsdf_submap_collection,
+      std::vector<MeshLayer::Ptr> *sub_map_mesh_layers);
+
+  // NOTE(alex.millane): Generates a mesh to test interpolation methods.
```

Comment:
I think it's time to archive the two functions below. The integrators appear to be working fine now.